subsys/net/lib/sockets/sockets_packet.c

```diff
@@ -78,7 +78,7 @@ static int zpacket_socket(int family, int type, int proto)
 	/* recv_q and accept_q are in union */
 	k_fifo_init(&ctx->recv_q);
 	zvfs_finalize_typed_fd(fd, ctx, (const struct fd_op_vtable *)&packet_sock_fd_op_vtable,
-			    ZVFS_MODE_IFSOCK);
+			       ZVFS_MODE_IFSOCK | ZVFS_MODE_CLOSE2);
```

Comment:
To me, this seems like it should be unnecessary

Reply:
I removed the extra flag, but this can be super confusing to a future developer looking at this code.
What I mean by this is that some of the networking code is now using `close(void*)` callback variant and some (currently `sockets_inet.c` and `sockets_packet.c`) are using the `close2(void*, int)` callback variant in order to fix the resource leak. In both cases the fdtable will call the `close2(void*, int)` if `IFSOCK` is set but the 2nd parameter is ignored most of the cases (I would have guessed compiler to give a warning in this case). For a future self looking this code in couple of years, this might probably look like a bug rather than a feature :confused:

Reply:
I think there will be a fair bit of refactorization here between now and a couple of years from now, mainly because FS also needs to get a bump in the near term for PSE52.

So consumers of `close()` can be upgraded to `close2()`, and the older version can be deprecated. Additionally, that would be a good time to introduce something like  a `k_off_t` type and deprecate the function signatures that use `ssize_t` and `off_t`.